syft/pkg/cataloger/static_binary/scan_binary.go

```diff
@@ -0,0 +1,228 @@
+package static_binary
+
+import (
+	"bytes"
+	"debug/elf"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/anchore/syft/internal/log"
+	"github.com/anchore/syft/syft/pkg"
+	"github.com/anchore/syft/syft/pkg/cataloger/internal/unionreader"
+)
+
+type fileInfo struct {
+	Name    string `json:"name"`
+	System  string `json:"system"`
+	Version string `json:"version"`
+}
+
+func scanFile(c *staticBinaryCataloger, reader unionreader.UnionReader, filename string) ([]byte, []string, error) {
+	bi, err := io.ReadAll(reader)
+	if err != nil {
+		log.WithFields("file", filename, "error", err).Trace("unable to read binary")
+		return bi, nil, err
+	}
+	//br is our byte reader
+	br := bytes.NewReader(bi)
+	//make a new elf file we can play with
+	e, err := elf.NewFile(br)
+	if e != nil {
+
+		symbols, err := e.ImportedLibraries()
+		if err != nil {
+			log.Debugf("unable to read elf binary: %s", err)
+			symbols = nil
+		}
+
+		//notes will be the raw data pulled from the section
+		noteSection := e.Section(".note.package")
+		if noteSection != nil {
+			notes, err := noteSection.Data()
+			var metadata pkg.StaticBinaryPackageMetadata
+			newerr := json.Unmarshal(notes, &metadata)
+			if newerr != nil {
+				log.Debugf("noteSection not nill, json didn't partse error")
+			}
+			if notes == nil {
+				log.Debugf("unable to read .note.package")
+			}
+
+			return notes, symbols, err
+		} else {
+			//This file did not have a note.package section. We must infer info another way
+			ver, system := fileNameTemplateVersionMatcher(c, filename)
+
+			//System is the same thing as namespace in the metadata we send over
+			data := fileInfo{
+				Name:    filename,
+				System:  system,
+				Version: ver,
+			}
+
+			b, _ := json.Marshal(data)
+
+			//fmt.Printf("ver: %v\n", filename)
+			return b, symbols, err
+		}
+	}
+	return nil, nil, err
+}
+
+func scanLibrary(c *staticBinaryCataloger, lib string) ([]byte, []string, error) {
+	var f *os.File
+	f, fileErr_ := getLibFile(c, lib)
+	if fileErr_ != nil {
+		fmt.Printf("Error: %v\n", fileErr_)
+	}
+	newUnionReader, _ := unionreader.GetUnionReader(f)
+	libnotes, liblibs, err := scanFile(c, newUnionReader, lib)
+	return libnotes, liblibs, err
+
+}
+
+func getLibFile(c *staticBinaryCataloger, symbol string) (*os.File, error) {
+	var f *os.File
+	var libPath string
+	//First we should check currect directory
+	libPath = "./"
+	f, fileErr_ := os.Open(libPath + symbol)
+	if fileErr_ != nil {
+		//Next we try lib64
+		libPath = "/lib64/"
+		f, fileErr_ = os.Open(libPath + symbol)
+		if fileErr_ != nil {
+			//then we try ld_library_path variable
+			libPath = os.Getenv("LD_LIBRARY_PATH")
```

Comment:
using environment variables directly within the cataloger isn't recommended, instead this should be passed in via a configuration struct, reducing side effects within catalogers

Reply:
Makes sense, we already have a config struct that lets user pass a path, so maybe we just remove this condition? The user options path overrides all this stuff.